app/src/main/java/com/example/detodito/Login.kt

```diff
@@ -0,0 +1,138 @@
+            startActivity(intent)
+            finish()
+        }
+
+        binding.loginButton.setOnClickListener {
+            val email = binding.emailEdittext.text.toString()
+            val password = binding.passwordEdittext.text.toString()
+
+            if (email.isNotEmpty() && password.isNotEmpty()) {
+                dialog.show()
+                try {
+                    auth.signInWithEmailAndPassword(email, password)
+                        .addOnCompleteListener(this) { task ->
+                            if (task.isSuccessful) {
+                                val user = FirebaseAuth.getInstance().currentUser
+                                user?.getIdToken(true)?.addOnCompleteListener { task ->
+                                    if (task.isSuccessful) {
+                                        val idToken = task.result?.token
+                                        if (idToken != null) {
+                                            shared(idToken, true)
+                                        }
+                                        val intent = Intent(applicationContext, MainActivity::class.java)
+                                        intent.putExtra("token", idToken)
+                                        dialog.dismiss()
+                                        startActivity(intent)
+                                        Toast.makeText(
+                                            baseContext, "Authentication Successful.",
+                                            Toast.LENGTH_SHORT
+                                        ).show()
+                                        finish()
+                                    } else {
+                                        dialog.dismiss()
+                                        Toast.makeText(
+                                            baseContext, "Authentication failed.",
+                                            Toast.LENGTH_SHORT
+                                        ).show()
+                                    }
+                                }
+                            } else {
+                                Toast.makeText(
+                                    baseContext, "Authentication failed.",
+                                    Toast.LENGTH_SHORT
+                                ).show()
+                            }
+                        }
+                } catch (e: Exception) {
+                    Toast.makeText(
+                        baseContext, "Authentication failed.",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    println(e.toString())
+                }
+            }
+
+            if (password.isEmpty()) {
+                // Set error text
+                binding.passwordLayout.error = "Ingresa una contraseña"
+            }
+
+            if (email.isEmpty()) {
+                binding.emailLayout.error = "Ingresa un correo valido"
+            }
+        }
```

Comment:
binding.loginButton.setOnClickListener {
    val email = binding.emailEdittext.text.toString()
    val password = binding.passwordEdittext.text.toString()

    // Verificar si los campos de email y contraseña están vacíos
    if (email.isEmpty()) {
        binding.emailLayout.error = "Ingresa un correo válido"
        return@setOnClickListener
    }

    if (password.isEmpty()) {
        binding.passwordLayout.error = "Ingresa una contraseña"
        return@setOnClickListener
    }

    // Mostrar el diálogo de carga
    dialog.show()

    // Iniciar sesión con email y contraseña
    try {
        auth.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                    val user = FirebaseAuth.getInstance().currentUser

                    // Obtener el token de autenticación del usuario
                    user?.getIdToken(true)?.addOnCompleteListener { task ->
                        if (task.isSuccessful) {
                            val idToken = task.result?.token

                            // Guardar el token en alguna parte
                            if (idToken != null) {
                                shared(idToken, true)
                            }

                            val intent = Intent(applicationContext, MainActivity::class.java)
                            intent.putExtra("token", idToken)

                            // Ocultar el diálogo de carga
                            dialog.dismiss()

                            // Iniciar la actividad principal
                            startActivity(intent)
                            Toast.makeText(
                                baseContext, "Autenticación exitosa.",
                                Toast.LENGTH_SHORT
                            ).show()

                            // Finalizar la actividad actual
                            finish()
                        } else {
                            // Ocultar el diálogo de carga y mostrar un mensaje de error
                            dialog.dismiss()
                            Toast.makeText(
                                baseContext, "Falló la autenticación.",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                } else {
                    // Mostrar un mensaje de error si la autenticación falla
                    Toast.makeText(
                        baseContext, "Falló la autenticación.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
    } catch (e: Exception) {
        // Mostrar un mensaje de error en caso de excepción
        Toast.makeText(
            baseContext, "Falló la autenticación.",
            Toast.LENGTH_SHORT
        ).show()
        println(e.toString())
    }
}

Reply:
Me parece bien el cambio